src/Symfony/Component/DependencyInjection/Loader/ContainerBuilderAwareLoader.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader;
+
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\Config\Loader\LoaderResolverInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Configuration loader that holds the ContainerBuilder inside.
+ *
+ * It can be useful to pass this to KernelInterface::registerContainerConfiguration()
+ * instead of a normal Loader if that method will need the ContainerBuilder.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class ContainerBuilderAwareLoader implements LoaderInterface
+{
+    /**
+     * @var ContainerBuilder
+     */
+    private $containerBuilder;
+
+    /**
+     * @var LoaderInterface
+     */
+    private $resourceLoader;
+
+    public function __construct(ContainerBuilder $builder, LoaderInterface $resourceLoader)
```

Comment:
Good point - and I can make this change if others agree. I'd prefer to push it off until we make that change. The main reason is so that `registerContainerConfiguration()` and `configureServices()`/`configureExtensions()` receive the exact same `LoaderInterface` object as an argument, making these two new methods look and act like what we're used to.